src/core/geometry/qgsgeometry.cpp

```diff
@@ -3174,6 +3174,34 @@ QgsGeometry QgsGeometry::forceRHR() const
   return forcePolygonClockwise();
 }
 
+Qgis::AngularDirection QgsGeometry::polygonOrientation() const
+{
+  if ( !d->geometry )
+  {
+    return Qgis::AngularDirection::Clockwise;
+  }
+
+  if ( isMultipart() )
+  {
+    const QgsGeometryCollection *collection = qgsgeometry_cast< const QgsGeometryCollection * >( d->geometry.get() );
+    const QgsAbstractGeometry *g = collection->geometryN( 0 );
+    if ( const QgsCurvePolygon *cp = qgsgeometry_cast< const QgsCurvePolygon * >( g ) )
+    {
+      return cp->exteriorRing()->orientation();
```

Comment:
I see at https://github.com/qgis/QGIS/blob/8c3e4d86bf1b9f35b1d95e4ce3db9611e9e5e577/src/core/geometry/qgsgeometry.cpp#L3063 that exteriorRing() return value is tested against nullptr

Reply:
done